From 959bcb4002f438e50cdef59d3b09b60e5ecf8a4c Mon Sep 17 00:00:00 2001
From: Harry Bagdi <harrybagdi@gmail.com>
Date: Tue, 30 Nov 2021 17:24:45 -0800
Subject: [PATCH 08/10] schema/init.lua: change uuid, regex and cjson usage

- Use an empty lua table for cjson.array_mt, this can potentially
introduce serilialization errors
- Use go.re2 instead of ngx.re
- Use go.uuid instead of uuid
---
 lua-tree/share/lua/5.1/kong/db/schema/init.lua | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/lua-tree/share/lua/5.1/kong/db/schema/init.lua b/lua-tree/share/lua/5.1/kong/db/schema/init.lua
index f5761c7..c265b80 100644
--- a/lua-tree/share/lua/5.1/kong/db/schema/init.lua
+++ b/lua-tree/share/lua/5.1/kong/db/schema/init.lua
@@ -1,12 +1,13 @@
 local tablex       = require "pl.tablex"
 local pretty       = require "pl.pretty"
 local utils        = require "kong.tools.utils"
-local cjson        = require "cjson"
+local cjson        = { array_mt = {} } --- TODO(hbagdi) XXX analyze the impact
+local uuid         = require "go.uuid".generate
 
 
 local setmetatable = setmetatable
-local re_match     = ngx.re.match
-local re_find      = ngx.re.find
+local re_match     = require "go.re2".match
+local re_find      = require "go.re2".find
 local concat       = table.concat
 local insert       = table.insert
 local format       = string.format
@@ -314,7 +315,7 @@ Schema.validators = {
     if #value ~= 36 then
       return nil
     end
-    return re_find(value, uuid_regex, "ioj") and true or nil
+    return re_find(value, uuid_regex) and true or nil
   end,
 
   contains = function(array, wanted)
-- 
2.25.1

